### DISCUSSION TITLE
Auto-Upgrading self hosted instances

### DISCUSSION THREAD

**Discussion by oddieis** [twentyhq/twenty#6798] [Ideas]
What plans are there in place to simplify upgrading a local instance whilst handling data migrations, and any custom objects created? It would be great to handle upgrades seamlessly with the click of a button, and support rollbacks too. Auto backups would be helpful in this scenario too.

**Comment by DevNoam** [comment]
+1 Very important features.
Tried to perform upgrade from 1.4 to 1.13 which failed at version 1.6. No matter the things I've tried to restore etc.. it just didn't let me upgrade after 1.6 because of data corruption. Trying to export data and importing is not possible as DB's not matching versioning.


These features are must:
- Auto upgrade instance: my .env conf is set to tag=latest, why can't it upgrade automatically (compose down, up, backup...).
- Option to export and Import all twenty data including workflows, settings tables... (This can be a good for exporting data from localhost to twenty managed cloud...)

Fortunately, these can be easily achieved using Linux automation and server side scripting, but without proper documentation and a standard rule set for handling auto updates with fault prevention, this makes it problematic.
